components/PoeModal.tsx

```diff
@@ -381,6 +385,45 @@ const PoeModal = () => {
     setAddress(undefined);
   };
 
+  const Checkboxes = () => (
+    <>
+      <div className="md-2 flex w-full items-center space-x-4 pt-8">
+        <label className="font-medium text-white">Sign immediately:</label>
+        <Checkbox
+          name="signImmediatelyFlag"
+          type="checkbox"
+          checked={signImmediatelyFlag}
+          onChange={async (e: ChangeEvent<HTMLInputElement>) => {
+            if (!e.target.checked) {
+              setResolveImmediatelyFlag(e.target.checked);
+            }
+            setSignImmediatelyFlag(e.target.checked);
```

Comment:
I didn't see anything that shows me that using a `useState` to store those value is needed. So instead you should use a `useRef`, it'll avoid rerendering needlessly the UI. 
Although, if there's a reason, please tell me and I'll be 100% fine with the `useState`

Reply:
I think that make sense mark both if select only resolve flag, so `useState` helps with it, make sense?